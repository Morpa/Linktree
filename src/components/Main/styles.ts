import styled, { css } from 'styled-components'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    text-align: center;
    align-items: center;
    justify-content: center;
    padding: ${theme.spacings.medium};
    color: ${theme.colors.white};
  `}
`

export const Logo = styled.img`
  ${({ theme }) => css`
    width: ${theme.spacings.xxxlarge};
    margin-bottom: ${theme.spacings.xxsmall};
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxxlarge};
    font-weight: ${theme.font.bold};
  `}
`

export const Description = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: ${theme.font.normal};
  `}
`
