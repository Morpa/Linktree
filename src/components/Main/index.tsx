import * as S from './styles'

const Main = ({
  title = 'Morpa',
  description = 'Welcome to my personal linktree'
}) => (
  <S.Wrapper>
    <S.Logo src="/img/morpa.svg" alt="Imagem de um cÃ­rculo com uma letra M" />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
  </S.Wrapper>
)

export default Main
