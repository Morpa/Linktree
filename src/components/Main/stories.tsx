import { Story, Meta } from '@storybook/react/types-6-0'
import { SectionHeaderProps } from 'types/api'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'title default',
    description: 'description default'
  }
} as Meta

export const Default: Story<SectionHeaderProps> = (args) => <Main {...args} />
