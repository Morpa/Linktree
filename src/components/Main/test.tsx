import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Main from '.'

const props = {
  title: 'Heading 1',
  description: 'Heading 2',
  image: 'Imagem de um c√≠rculo com uma letra M'
}

describe('<Main />', () => {
  it('should render main with title, description and image', () => {
    const { container } = renderWithTheme(<Main {...props} />)

    expect(
      screen.getByRole('heading', { name: /heading 1/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /heading 2/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: props.image })).toHaveAttribute(
      'src',
      '/img/morpa.svg'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
