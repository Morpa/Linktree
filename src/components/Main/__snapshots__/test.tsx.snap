// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render main with title, description and image 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3.2rem;
  color: #FAFAFA;
}

.c1 {
  width: 15rem;
  margin-bottom: 0.8rem;
}

.c2 {
  font-size: 4.5rem;
  font-weight: 600;
}

.c3 {
  font-size: 2.0rem;
  font-weight: 400;
}

<main
  class="c0"
>
  <img
    alt="Imagem de um cÃ­rculo com uma letra M"
    class="c1"
    src="/img/morpa.svg"
  />
  <h1
    class="c2"
  >
    Heading 1
  </h1>
  <h3
    class="c3"
  >
    Heading 2
  </h3>
</main>
`;
